stack/dispatcher.go

```diff
@@ -0,0 +1,104 @@
+package stack
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tendermint/tmlibs/log"
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/errors"
+	"github.com/tendermint/basecoin/types"
+)
+
+const (
+	NameDispatcher = "disp"
+)
+
+// Dispatcher grabs a bunch of Dispatchables and groups them into one Handler.
+//
+// It will route tx to the proper locations and also allows them to call each
+// other synchronously through the same tx methods.
+type Dispatcher struct {
+	routes map[string]Dispatchable
```

Comment:
maps are a non-deterministic type as I learned the hard way - not sure if this is okay here or not, just a heads up